onchain/rollups/test/foundry/consensus/quorum/QuorumFactory.t.sol

```diff
@@ -0,0 +1,121 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+/// @title Quorum Factory Test
+pragma solidity ^0.8.8;
+
+import {Test} from "forge-std/Test.sol";
+import {QuorumFactory} from "contracts/consensus/quorum/QuorumFactory.sol";
+import {Quorum} from "contracts/consensus/quorum/Quorum.sol";
+import {IHistory} from "contracts/history/IHistory.sol";
+import {Vm} from "forge-std/Vm.sol";
+
+contract QuorumFactoryTest is Test {
+    QuorumFactory factory;
+    address[] validators;
+    uint256[] shares;
+    uint256 numValidator;
+
+    event QuorumCreated(Quorum quorum, IHistory history);
+
+    function setUp() public {
+        factory = new QuorumFactory();
+    }
+
+    function testNewQuorum(uint16 _numValidator, IHistory _history) public {
```

Comment:
We can have `_numValidator` as a `uint256` fuzzy parameter, and use [`bound`](https://book.getfoundry.sh/reference/forge-std/bound) from `forge-std` to restrict its domain to the desired $(0,1000)$ range. This should also improve performance of this fuzzy test case, as the function will go from a $998/2^{16} \approx 1.5\\%$ chance of passing through the `vm.assume` check to $100\\%$ (no `vm.assume`).